### PR TITLE
fix: support unary plus and minus for case

### PR DESCRIPTION
**Pull request by polvalente**

### CODE REVIEW

File: nx/lib/nx/shared.ex [hunk 1]
```diff
@@ -290,8 +290,8 @@ defmodule Nx.Shared do
       defmacro unquote(name)(unquote_splicing(args)) do
         {module, name} =
           case __CALLER__.context do
-            :guard -> {Kernel, unquote(name)}
-            _ -> {__MODULE__, unquote(fallback)}
+            ctx when ctx not in [:guard, :match] -> {__MODULE__, unquote(fallback)}
```
Comment (polvalente) [inline nx/lib/nx/shared.ex:293]: As I'm not sure which values are valid, I chose to explicitly allow :guard and :match only

**Review by josevalim — APPROVED** [review]

File: nx/test/nx/defn_test.exs [hunk 2]
```diff
@@ -243,6 +243,26 @@ defmodule Nx.DefnTest do
   describe "unary ops" do
     defn exp(t), do: Nx.exp(t)
 
+    defn unary_plus_minus_guards(t, opts \\ []) do
+      case opts[:value] do
+        value when value == -2 ->
+          Nx.add(t, value)
```
Comment (josevalim) [inline nx/test/nx/defn_test.exs:249]: Make this return -1 or 1 in order to simplify the tests. See the tests I have added for `case` itself.

**Review by josevalim — CHANGES_REQUESTED** [review]
Let's simplify the tests a bit!

File: nx/lib/nx/shared.ex (hunk 1 above)
Comment (josevalim) [reply nx/lib/nx/shared.ex]: it is only guard/match/nil.

**Review by josevalim — APPROVED** [review]

**Pull request merged**